src/Features/CSharp/Portable/Completion/CompletionProviders/EnumAndCompletionListTagCompletionProvider.cs

```diff
@@ -61,102 +66,118 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
-                    {
-                        continue;
-                    }
+                type = enumType;
+            }
 
-                    // Does type have any aliases?
-                    var alias = await type.FindApplicableAliasAsync(position, semanticModel, cancellationToken).ConfigureAwait(false);
+            var options = context.Options;
+            var hideAdvancedMembers = options.GetOption(CompletionOptions.HideAdvancedMembers, semanticModel.Language);
+            if (!type.IsEditorBrowsable(hideAdvancedMembers, semanticModel.Compilation))
+                return;
 
-                    var displayText = alias != null
-                        ? alias.Name
-                        : type.ToMinimalDisplayString(semanticModel, position);
+            // Does type have any aliases?
+            var alias = await type.FindApplicableAliasAsync(position, semanticModel, cancellationToken).ConfigureAwait(false);
 
-                    var workspace = document.Project.Solution.Workspace;
-                    var text = await semanticModel.SyntaxTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var displayText = alias != null
+                ? alias.Name
+                : type.ToMinimalDisplayString(semanticModel, position);
 
-                    var item = SymbolCompletionItem.CreateWithSymbolId(
-                        displayText: displayText,
-                        displayTextSuffix: "",
-                        symbols: ImmutableArray.Create(alias ?? type),
-                        rules: s_rules.WithMatchPriority(MatchPriority.Preselect),
-                        contextPosition: position);
+            // Add the enum itself.
+            var symbol = alias ?? type;
+            var sortText = symbol.Name;
 
-                    context.AddItem(item);
-                }
-            }
-            catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e))
+            context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                displayText,
+                displayTextSuffix: "",
+                symbols: ImmutableArray.Create(symbol),
+                rules: s_enumTypeRules,
+                contextPosition: position,
+                sortText: sortText));
+
+            // And now all the accessible members of the enum.
+            if (type.TypeKind == TypeKind.Enum)
             {
-                throw ExceptionUtilities.Unreachable;
+                // We'll want to build a list of the actual enum members and all accessible instances of that enum, too
+                var index = 0;
+
+                var fields = type.GetMembers().OfType<IFieldSymbol>().Where(f => f.IsConst).Where(f => f.HasConstantValue);
+                foreach (var field in fields.OrderBy(f => IntegerUtilities.ToInt64(f.ConstantValue)))
```

Comment:
Sorting fails for enums backed by a  `ulong` with values > `Int64.MaxValue`. @CyrusNajmabadi Do you want me to fix this?

Reply:
Feels unlikely to matter in practice. But if you'd like, go ahead.